src/acceptance-test/java/org/zalando/nakadi/webservice/timelines/TimelineSyncAT.java

```diff
@@ -0,0 +1,201 @@
+            uuidGenerator = new UUIDGenerator();
+        }
+        if (null == zookeeperHolder) {
+            zookeeperHolder = Mockito.mock(ZooKeeperHolder.class);
+            Mockito.when(zookeeperHolder.get()).thenReturn(CURATOR);
+        }
+        final TimelineSyncImpl result = new TimelineSyncImpl(zookeeperHolder, uuidGenerator);
+        synchronized (DELAYED_RUNS) {
+            DELAYED_RUNS.add(() -> {
+                try {
+                    result.reactOnEventTypesChange();
+                } catch (final InterruptedException ex) {
+                    Thread.currentThread().interrupt();
+                }
+            });
+        }
+        return result;
+    }
+
+    @BeforeClass
+    public static void initialize() {
+        delayedRunsExecutor = new Thread(() -> {
+            while (true) {
+                try {
+                    Thread.sleep(300L);
+                    synchronized (DELAYED_RUNS) {
+                        DELAYED_RUNS.forEach(Runnable::run);
+                    }
+                } catch (final InterruptedException e) {
+                    break;
+                }
+            }
+        }, "AT_delayed_runs");
+        delayedRunsExecutor.setDaemon(true);
+        delayedRunsExecutor.start();
+    }
+
+    @AfterClass
+    public static void terminate() {
+        delayedRunsExecutor.interrupt();
+        delayedRunsExecutor = null;
+        CURATOR.close();
+    }
+
+    @Test
+    public void testNodeInformationWrittenOnStart() throws Exception {
+        final TimelineSync sync = createTimeline();
+        final String currentVersion = new String(CURATOR.getData().forPath("/nakadi/timelines/version"));
+        Assert.assertEquals(currentVersion,
+                new String(CURATOR.getData().forPath("/nakadi/timelines/nodes/" + sync.getNodeId())));
+    }
+
+    @Test
+    public void testTimelineUpdateWaitsForActivePublish() throws InterruptedException, IOException, TimeoutException {
+        final TimelineSync t1 = createTimeline();
+        final TimelineSync t2 = createTimeline();
+        final String eventType = UUID.randomUUID().toString();
+        final AtomicBoolean updated = new AtomicBoolean(false);
+
+        final Closeable publishing = t1.workWithEventType(eventType, TimeUnit.SECONDS.toMillis(1));
+        try {
+            new Thread(() -> {
+                try {
+                    t2.startTimelineUpdate(eventType, TimeUnit.SECONDS.toMillis(30));
+                    updated.set(true);
+                } catch (final InterruptedException ex) {
+                    throw new RuntimeException(ex);
+                }
+            }, "timeline_update").start();
+            // Wait a little bit for thread to start timeline update
+            Thread.sleep(TimeUnit.SECONDS.toMillis(3));
+            Assert.assertEquals(false, updated.get());
+        } finally {
+            publishing.close();
+        }
+        int attemptsLeft = 5;
+        while (attemptsLeft-- > 0) {
```

Comment:
Have you considered using existent retry strategies like Retryer.executeWithRetry?